src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1268,6 +1286,26 @@ bool entitiesQuery
 
                 cer->contextElement.contextAttributeVector.push_back(caP);
             }
+        }        
+
+        /* All the attributes existing in the request but not found in the response are added with 'found' set to false */
+        for (unsigned int ix = 0; ix < attrL.size(); ++ix)
+        {
+          bool found = false;
+          std::string attrName = attrL.get(ix);
+          for (unsigned int jx = 0; jx < cer->contextElement.contextAttributeVector.size(); ++jx)
+          {
+            if (attrName == cer->contextElement.contextAttributeVector.get(jx)->name)
+            {
+              found = true;
+              break;
+            }
+          }
+          if (!found)
+          {
+            ContextAttribute* caP = new ContextAttribute(attrName, "", "", false);
+            cer->contextElement.contextAttributeVector.push_back(caP);
+          }
```

Comment:
There are a few lines to save here also (4) - if you skip the variable 'found' and instead of setting found = true in the 'if-body' you just create and push the caP to cer->contextElement.contextAttributeVector.
The second if is no longer necessary, of course.